app/models/template_input.rb

```diff
@@ -95,7 +103,13 @@ def ready?
     end
 
     def resolved_value
-      input_value.value
+      value = input_value.value
+
+      if @input.options_array.blank? || @input.options_array.include?(value)
+        value
+      else
+        raise(ValueNotPermitted, "Value '#{value}' is not a valid selection for #{@input.name}")
```

Comment:
And if permissions are only evaluated at scheduling then the behavior is fine, I'll check check the values in a normal rails callback.